src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -302,6 +419,131 @@ void VST3Wrapper::Finalize()
    mEffectComponent->setActive(false);
 }
 
+void VST3Wrapper::ConsumeChanges(EffectSettings& settings)
+{
+   const auto& vst3settings = GetSettings(settings);
+   for(auto& p : vst3settings.parameterChanges)
+   {
+      auto it = std::find_if(mParameters.begin(), mParameters.end(), [&p](const auto& v) { return v.first == p.first; });
+      if(it != mParameters.end())
+         it->second = p.second;
+      else
+         mParameters.push_back(p);
+   }
```

Comment:
Should this function clear the map in the settings? -- maybe not if it needs to be done in a worker thread.

You don't (yet) use it in any worker thread (at the eighth commit of ten).

Reply:
At this commit I see accumulation of changes in the map in `performEdit` but clearing nowhere.